app/controllers/application_controller.rb

```diff
@@ -1,10 +1,149 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
+  
+  register Sinatra::ActiveRecordExtension
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "its_a_secret"
   end
 
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if !is_logged_in
+      erb :'registrations/signup'
+    else
+      redirect '/tweets'
+    end 
+  end
+
+  post '/signup' do
+    if params[:username] != '' && params[:password] != '' && params[:email] != ''
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) to perform these checks more easily.